components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/services/UserInformationRecoveryService.java

```diff
@@ -814,7 +815,7 @@ public VerificationBean registerUser(String userName, String password,
             IdentityMgtConfig config = IdentityMgtConfig.getInstance();
 
             if (isListenerEnable && config.isAuthPolicyAccountLockOnCreation()) {
-                UserDTO userDTO = new UserDTO(userName);
+                UserDTO userDTO = new UserDTO(UserCoreUtil.addTenantDomainToEntry(userName, tenantDomain));
```

Comment:
Looks like in many other places we use Utils.processUserId() method to build the username. Can we use the same here. This particular util method checks for SaaS application config and all. Therefore we must use this.